api/customers/customers.go

```diff
@@ -25,14 +42,42 @@ func NewCustomerHandler(jwt *tokens.JWT, db *sql.DB) *CustomerHandler {
 	return &CustomerHandler{jwt, db}
 }
 
-type RegisterResponse struct {
-	Status string `json:"status"`
-}
+func (ch *CustomerHandler) GetProfile(w http.ResponseWriter, r *http.Request) {
+	token := ch.jwt.GetToken(r)
+	err := token.Valid()
+	if err != nil {
+		helpers.HTTPError(w, http.StatusBadRequest, err.Error())
+		return
+	}
 
-type EmailResponse struct {
-	Email string `json:"email"`
+	cus, err := models.GetProfile(ch.db, token.CustId)
+	if err != nil {
+		helpers.HTTPError(w, http.StatusBadRequest, err.Error())
+		return
+	}
+
+	acc, err := models.GetMainAccount(ch.db, cus.AccountNum)
+	if err != nil {
+		helpers.HTTPError(w, http.StatusBadRequest, err.Error())
+		return
+	}
+
+	result := GetProfileResult{
+		Customers: cus,
+		Accounts:  acc,
+	}
+
+	res, err := json.Marshal(result)
+	if err != nil {
+		helpers.HTTPError(w, http.StatusBadRequest, err.Error())
+		return
+	}
+
+	fmt.Fprintln(w, string(res))
 }
 
+var emailRegex = regexp.MustCompile("^[a-zA-Z0-9.!#$%&'*+\\/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$")
```

Comment:
you can move emailRegex into inside isEmailValid function because u dont need this globally